dts/arm/silabs/efm32_jg_pg_12b.dtsi

```diff
@@ -132,75 +132,93 @@
 			#size-cells = <1>;
 
 			gpioa: gpio@4000a000 {
-				compatible = "silabs,efm32-gpio-port";
+				compatible = "silabs,gecko-gpio-port";
 				reg = <0x4000a000 0x30>;
+				peripheral-id = <0>;
 				label = "GPIO_A";
 				gpio-controller;
 				#gpio-cells = <2>;
+				status = "disabled";
 			};
 
 			gpiob: gpio@4000a030 {
-				compatible = "silabs,efm32-gpio-port";
+				compatible = "silabs,gecko-gpio-port";
 				reg = <0x4000a030 0x30>;
+				peripheral-id = <1>;
 				label = "GPIO_B";
 				gpio-controller;
 				#gpio-cells = <2>;
+				status = "disabled";
 			};
 
 			gpioc: gpio@4000a060 {
-				compatible = "silabs,efm32-gpio-port";
+				compatible = "silabs,gecko-gpio-port";
 				reg = <0x4000a060 0x30>;
+				peripheral-id = <2>;
 				label = "GPIO_C";
 				gpio-controller;
 				#gpio-cells = <2>;
+				status = "disabled";
 			};
 
 			gpiod: gpio@4000a090 {
-				compatible = "silabs,efm32-gpio-port";
+				compatible = "silabs,gecko-gpio-port";
 				reg = <0x4000a090 0x30>;
+				peripheral-id = <3>;
 				label = "GPIO_D";
 				gpio-controller;
 				#gpio-cells = <2>;
+				status = "disabled";
 			};
 
 			gpioe: gpio@4000a0c0 {
-				compatible = "silabs,efm32-gpio-port";
+				compatible = "silabs,gecko-gpio-port";
 				reg = <0x4000a0c0 0x30>;
+				peripheral-id = <4>;
 				label = "GPIO_E";
 				gpio-controller;
 				#gpio-cells = <2>;
+				status = "disabled";
 			};
 
 			gpiof: gpio@4000a0f0 {
-				compatible = "silabs,efm32-gpio-port";
+				compatible = "silabs,gecko-gpio-port";
 				reg = <0x4000a0f0 0x30>;
+				peripheral-id = <5>;
 				label = "GPIO_F";
 				gpio-controller;
 				#gpio-cells = <2>;
+				status = "disabled";
 			};
 
 			gpioi: gpio@4000a180 {
-				compatible = "silabs,efm32-gpio-port";
+				compatible = "silabs,gecko-gpio-port";
 				reg = <0x4000a180 0x30>;
+				peripheral-id = <8>;
 				label = "GPIO_I";
 				gpio-controller;
 				#gpio-cells = <2>;
+				status = "disabled";
 			};
 
-			gpioj: gpio@4000a1c0 {
-				compatible = "silabs,efm32-gpio-port";
-				reg = <0x4000a1c0 0x30>;
+			gpioj: gpio@4000a1b0 {
```

Comment:
Register address modification is a bugfix. I should have submitted it in a separate commit, will do if requested.